### DISCUSSION TITLE
toggle snapcast stream

### DISCUSSION THREAD

**Discussion by zalmyr** [CarlosDerSeher/snapclient#26] [General]
I'm curious, what it would take to add button to the esp32 that can toggle between different snapcast streams.

**Comment by CarlosDerSeher** [comment]
You'll need a http client to send commands to snapserver. Shouldn't be too hard to add though, there are plenty of examples how to do http client for esp32.

The hard part for me would be to find the correct josn messages to send through snapserver control interface to do what you suggest. I don't know enough about it to be of much help here.

You also could use raw TCP Sockets and probably need something documented around here: https://github.com/badaix/snapcast/blob/develop/doc/json_rpc_api/control.md#groupsetstream
